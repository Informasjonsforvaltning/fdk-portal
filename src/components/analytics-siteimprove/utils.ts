export enum EventCategory {
  DETAILS_PAGE = 'Details page'
}

export enum EventAction {
  REQUEST_ACCESS = 'Request access'
}

type SiteImproveEventProps = {
  category: EventCategory;
  action: EventAction;
  label?: string | undefined;
};

export const trackSiteImproveEvent = ({
  category,
  action,
  label
}: SiteImproveEventProps) => {
  if (window._sz === undefined) {
    // eslint-disable-next-line no-console
    console.error('Unable to find Site Improve event library.');
  } else if (label) {
    window._sz.push(['event', category, action, label]);
  } else {
    window._sz.push(['event', category, action]);
  }
};
