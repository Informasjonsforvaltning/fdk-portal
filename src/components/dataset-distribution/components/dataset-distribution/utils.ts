import { DataFormat } from '../../../../types/enums';

const priorities: { [key: string]: number } = {
  [DataFormat.JSON]: 1,
  [DataFormat.XML]: 2,
  [DataFormat.CSV]: 3
};

export const formatSorter = (a: DataFormat, b: DataFormat): number => {
  if (!(a in priorities)) {
    return 1;
  }
  if (!(b in priorities)) {
    return -1;
  }
  return priorities[a] > priorities[b] ? 1 : -1;
};
