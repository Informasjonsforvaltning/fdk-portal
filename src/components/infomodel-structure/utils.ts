import { InformationModelProperty, TextLanguage } from '../../types';

const applyRelationalTitle = (
  hasObjectType?: string | null,
  title?: Partial<TextLanguage> | null
) => {
  const parentObject = hasObjectType?.split('#').pop();
  const objectString = parentObject
    ? `${parentObject.charAt(0).toUpperCase() + parentObject.slice(1)}: `
    : '';
  return title
    ? Object.entries(title).reduce(
        (previous, [key, value]) => ({
          ...previous,
          [key]: objectString + value
        }),
        {} as Partial<TextLanguage>
      )
    : title;
};

export const resolvePropertySymmetries = (
  modelProperties: Record<string, Partial<InformationModelProperty>>
) =>
  Object.entries(modelProperties).reduce((previous, [key, value]) => {
    if (value.formsSymmetryWith) {
      return {
        ...previous,
        [key]: {
          ...value,
          title: applyRelationalTitle(
            modelProperties[value.formsSymmetryWith].hasObjectType,
            value.title
          )
        },
        [value.formsSymmetryWith]: {
          ...modelProperties[value.formsSymmetryWith],
          formsSymmetryWith: key,
          title: applyRelationalTitle(
            value.hasObjectType,
            modelProperties[value.formsSymmetryWith].title
          )
        }
      };
    }

    return { ...previous, [key]: value };
  }, {} as Record<string, Partial<InformationModelProperty>>);
