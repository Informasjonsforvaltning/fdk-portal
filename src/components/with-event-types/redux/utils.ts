import { Namespace, Fetcher, graph, isLiteral, Literal } from 'rdflib';

import type { EventType } from '../../../types';
import { getConfig } from '../../../config';

const rdf = Namespace('http://www.w3.org/1999/02/22-rdf-syntax-ns#');
const skos = Namespace('http://www.w3.org/2004/02/skos/core#');

export const getEventTypes = async (): Promise<EventType[]> => {
  const fetcher = new Fetcher(graph(), {});

  await fetcher.load(`${getConfig().searchHost.host}/api/events`, {
    withCredentials: false
  });

  const eventTypes = fetcher.store.statementsMatching(
    null,
    rdf('type'),
    skos('Concept')
  );

  return eventTypes.map((item: any) => ({
    uri: item.subject.value ?? '',
    prefLabel: fetcher.store
      .statementsMatching(item.subject, skos('prefLabel'))
      .map(({ object }) => object as Literal)
      .filter(isLiteral)
      .reduce(
        (previous, { value, language }) => ({
          ...previous,
          [language]: value
        }),
        {}
      ),
    broader: fetcher.store
      .statementsMatching(item.subject, skos('broader'))
      .map(({ object }) => object.value)
  }));
};
