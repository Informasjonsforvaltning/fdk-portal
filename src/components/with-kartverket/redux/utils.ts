import {
  Namespace,
  Fetcher,
  graph,
  isLiteral,
  isNamedNode,
  Literal
} from 'rdflib';

import type { AdministrativeUnit } from '../../../types';
import { AdministrativeUnitType } from '../../../types/enums';

const owl = Namespace('http://www.w3.org/2002/07/owl#');
const rdf = Namespace('http://www.w3.org/1999/02/22-rdf-syntax-ns#');
const rdfs = Namespace('http://www.w3.org/2000/01/rdf-schema#');

export const getAdministrativeUnit = async (
  uri: string
): Promise<AdministrativeUnit> => {
  const fetcher = new Fetcher(graph(), { withCredentials: false });

  await fetcher.load(
    `https://rdf.kartverket.no/api/1.0/adminstrative_unit/describe?uri=${uri}&mode=CBD&format=text/turtle`,
    {}
  );

  const subject = fetcher.store.anyStatementMatching(
    null,
    rdf('type'),
    owl('NamedIndividual')
  )?.subject;

  return {
    uri: subject?.value ?? '',
    name: fetcher.store
      .statementsMatching(subject, rdfs('label'))
      .map(({ object }) => object as Literal)
      .filter(isLiteral)
      .reduce(
        (previous, { value, language }) => ({
          ...previous,
          [language]: value
        }),
        {}
      ),
    type: fetcher.store
      .statementsMatching(subject, rdf('type'))
      .map(({ object }) => object)
      .filter(isNamedNode)
      .filter(({ value }) =>
        Object.values(AdministrativeUnitType).includes(value as any)
      )
      .pop()?.value as AdministrativeUnitType
  };
};
