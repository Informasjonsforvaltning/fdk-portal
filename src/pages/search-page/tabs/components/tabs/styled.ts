import styled from 'styled-components';

import AllIconBase from '../../../../../images/icon-catalog-all-md.svg';
import DatasetIconBase from '../../../../../images/icon-catalog-dataset-md.svg';
import ApiIconBase from '../../../../../images/icon-catalog-api-md.svg';
import ConceptIconBase from '../../../../../images/icon-catalog-concept-md.svg';
import InfomodIconBase from '../../../../../images/icon-catalog-infomod-md.svg';

const Tabs = styled.ul`
  align-items: center;
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
  margin-bottom: 0;
  padding-left: 0;
  @media (max-width: 768px) {
    justify-content: space-around;
    margin: 0 10%;
  }
`;

const Label = styled.span`
  margin-left: 0.5em;
  @media (max-width: 768px) {
    display: none;
    margin-left: 0;
  }
`;

const MobileLabel = styled.div`
  background-color: #eeeff0;
  display: flex;
  justify-content: center;
  font-size: 2rem;
  padding-top: 1em;
  @media (min-width: 768px) {
    display: none;
  }
`;

const AllIcon = styled(AllIconBase)`
  height: 1.5em;
`;

const DatasetIcon = styled(DatasetIconBase)`
  border-radius: 50%;
  padding: 0.4em;
  background-color: ${({ theme }) => theme.colors.dataset.light};
  height: 2em;
  fill: ${({ theme }) => theme.colors.dataset.dark};
`;

const ApiIcon = styled(ApiIconBase)`
  border-radius: 50%;
  padding: 0.4em;
  background-color: ${({ theme }) => theme.colors.api.light};
  height: 2em;
  fill: ${({ theme }) => theme.colors.api.dark};
`;

const ConceptIcon = styled(ConceptIconBase)`
  border-radius: 50%;
  padding: 0.4em;
  background-color: ${({ theme }) => theme.colors.concept.light};
  height: 2em;
  fill: ${({ theme }) => theme.colors.concept.dark};
`;

const InfomodIcon = styled(InfomodIconBase)`
  border-radius: 50%;
  padding: 0.4em;
  background-color: ${({ theme }) => theme.colors.infomod.light};
  height: 2em;
  fill: ${({ theme }) => theme.colors.infomod.dark};
`;

export default {
  Tabs,
  Label,
  MobileLabel,
  AllIcon,
  DatasetIcon,
  ApiIcon,
  ConceptIcon,
  InfomodIcon
};
