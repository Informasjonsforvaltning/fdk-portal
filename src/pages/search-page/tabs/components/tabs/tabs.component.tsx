import React, { FC, memo } from 'react';
import { RouteComponentProps, withRouter } from 'react-router';

import SC from './styled';
import Tab from '../tab/tab.component';
import localization from '../../../../../lib/localization';
import {
  PATHNAME_APIS,
  PATHNAME_CONCEPTS,
  PATHNAME_DATASETS,
  PATHNAME_INFORMATIONMODELS
} from '../../../../../constants/constants';
import { getLinkForTab } from '../../../search-location-helper';

interface Props {
  countDatasets: number;
  countConcepts: number;
  countApis: number;
  countInformationModels: number;
}

const Tabs: FC<Props & RouteComponentProps> = ({
  countDatasets,
  countApis,
  countConcepts,
  countInformationModels,
  location
}) => {
  const renderLabel = () => {
    switch (location.pathname) {
      case PATHNAME_DATASETS: {
        return localization.formatString(
          localization.showHitsFor,
          `${countDatasets}`,
          localization.datasetLabel.toLowerCase()
        );
      }
      case PATHNAME_APIS: {
        return localization.formatString(
          localization.showHitsFor,
          `${countApis}`,
          localization.apiLabel.toLowerCase()
        );
      }
      case PATHNAME_CONCEPTS: {
        return localization.formatString(
          localization.showHitsFor,
          `${countConcepts}`,
          localization.conceptLabel.toLowerCase()
        );
      }
      case PATHNAME_INFORMATIONMODELS: {
        return localization.formatString(
          localization.showHitsFor,
          `${countInformationModels}`,
          localization.informationModelLabel.toLowerCase()
        );
      }
      default: {
        return null;
      }
    }
  };
  return (
    <>
      <SC.Tabs>
        <Tab
          active={location.pathname === PATHNAME_DATASETS}
          tabLink={getLinkForTab(location, PATHNAME_DATASETS)}
          label={localization.datasetLabel}
        >
          <SC.DatasetIcon />
          <SC.Label>
            {localization.page.datasetTab}&nbsp;({countDatasets})
          </SC.Label>
        </Tab>

        <Tab
          active={location.pathname === PATHNAME_APIS}
          tabLink={getLinkForTab(location, PATHNAME_APIS)}
          label={localization.apiLabel}
        >
          <SC.ApiIcon />
          <SC.Label>
            {localization.page.apiTab}&nbsp;({countApis})
          </SC.Label>
        </Tab>

        <Tab
          active={location.pathname === PATHNAME_CONCEPTS}
          tabLink={getLinkForTab(location, PATHNAME_CONCEPTS)}
          label={localization.conceptLabel}
        >
          <SC.ConceptIcon />
          <SC.Label>
            {localization.page.termTab}&nbsp;({countConcepts})
          </SC.Label>
        </Tab>

        <Tab
          active={location.pathname === PATHNAME_INFORMATIONMODELS}
          tabLink={getLinkForTab(location, PATHNAME_INFORMATIONMODELS)}
          label={localization.informationModelLabel}
        >
          <SC.InfomodIcon />
          <SC.Label>
            {localization.page.informationModelTab}&nbsp;(
            {countInformationModels})
          </SC.Label>
        </Tab>
      </SC.Tabs>
      <SC.MobileLabel>{renderLabel()}</SC.MobileLabel>
    </>
  );
};

export default withRouter(memo(Tabs));
