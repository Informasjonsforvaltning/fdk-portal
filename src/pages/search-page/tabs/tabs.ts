/* istanbul ignore file */
import TabSC from './components/tab/styled';

export { default as Tabs } from './components/tabs/tabs.component';
export { default as Tab } from './components/tab/tab.component';

export const SC = {
  Tab: TabSC
};
