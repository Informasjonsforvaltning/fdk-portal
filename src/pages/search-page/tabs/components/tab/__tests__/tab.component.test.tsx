/**
 * @jest-environment jsdom
 */
import React from 'react';
import { cleanup } from '@testing-library/react';
import { shallow } from 'enzyme';

import { Expectation } from '../../../../../../../test/utils';
import { Tab } from '../../../tabs';

afterEach(cleanup);

describe('Tab component', () => {
  const tabLink = { pathname: '/datasets', search: '' };
  const label = 'Tab for datasett';
  it(Expectation.STRUCTURE, () => {
    const container = shallow(<Tab active tabLink={tabLink} label={label} />);
    expect(container).toMatchSnapshot();
  });
  it(Expectation.STRUCTURE, () => {
    const container = shallow(
      <Tab active={false} tabLink={tabLink} label={label} />
    );
    expect(container).toMatchSnapshot();
  });
});
