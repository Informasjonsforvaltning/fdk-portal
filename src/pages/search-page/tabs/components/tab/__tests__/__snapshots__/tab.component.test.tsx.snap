// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab component must render correct structure 1`] = `
<styled.li
  active={true}
>
  <Styled(Link)
    aria-label="Tab for datasett"
    to={
      Object {
        "pathname": "/datasets",
        "search": "",
      }
    }
  />
</styled.li>
`;

exports[`Tab component must render correct structure 2`] = `
<styled.li
  active={false}
>
  <Styled(Link)
    aria-label="Tab for datasett"
    to={
      Object {
        "pathname": "/datasets",
        "search": "",
      }
    }
  />
</styled.li>
`;
